mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -63,24 +65,75 @@ pub struct NvmeIoChannelInner<'a> {
     pub qpair: *mut spdk_nvme_qpair,
     poll_group: NonNull<spdk_nvme_poll_group>,
     poller: poller::Poller<'a>,
+    // Flag to indicate the shutdown state of the channel.
+    // We need such a flag to differentiate between channel reset and shutdown.
+    // Channel reset is a reversible operation, which is followed by
+    // reinitialize(), which 'resurrects' channel (i.e. recreates all its
+    // I/O resources): such behaviour is observed during controller reset.
+    // Shutdown, in contrary, means 'one-way' ticket for the channel, which
+    // doesn't assume any further resurrections: such behaviour is seen
+    // upon controller shutdown. Being able to differentiate between these
+    // 2 states allows controller reset to behave properly in parallel with
+    // shutdown (if case reset is initiated before shutdown), and
+    // not to reinitialize channels already processed by shutdown logic.
+    is_shutdown: bool,
```

Comment:
If it's false, do we need differentiate between reset and not reset? If so should this be an enum?

Reply:
There is no need to differentiate between reset/shutdown when it's false. We just need a flag telling whether recover for the channel is possible or not. There is no complex logic around it.